### PR TITLE
feat: signed state file manager wiring cleanup

### LINKED ISSUE #13323: Improve signed state file manager wiring
Improve how the signed state file manager is wired.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13323

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 1]
```diff
@@ -1031,4 +1034,41 @@
         }
         return stateHasher;
     }
+
+    /**
+     * Provide a state snapshot manager in place of the platform's default state snapshot manager.
+     *
+     * @param stateSnapshotManager the state file manager to use
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1041]: nit: doc references old naming `file manager`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 2]
```diff
@@ -1031,4 +1034,41 @@ public StateHasher buildStateHasher() {
...
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withStateSnaphsotManager(@NonNull final StateSnapshotManager stateSnapshotManager) {
+        throwIfAlreadyUsed();
+        if (this.stateSnapshotManager != null) {
+            throw new IllegalStateException("State file manager has already been set");
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1048]: "state file manager" -> "state snapshot manager"

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java [hunk 3]
```diff
@@ -159,10 +158,12 @@ public void savedStateInFreezePeriod() {
     }
 
     /**
-     * {@inheritDoc}
+     * Loads all necessary data from the {@code reservedSignedState}.
+     *
+     * @param signedState
+     * 		the signed state to load
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java:164]: nit: doc has old tab based formatting

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/DefaultStateSnapshotManager.java [hunk 4]
```diff
@@ -117,7 +124,9 @@ public SignedStateFileManager(
      *                            this method call and this method will release the reservation when it is done
      * @return the result of the state saving operation, or null if the state was not saved
      */
-    public @Nullable StateSavingResult saveStateTask(@NonNull final ReservedSignedState reservedSignedState) {
+    @Override
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/DefaultStateSnapshotManager.java:127]: The methods in the class are now `@Override`s, and duplicate the javadocs that exist in the `StateSnapshotManager` interface

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            .withDescription("average time it takes to do perform all actions when writing a SignedState to disk "
+                    + "(in milliseconds)")
+            .withUnit("ms");
+    private final RunningAverageMetric stateToDiskTime;
+
+    private static final Counter.Config TOTAL_NEVER_SIGNED_STATES_CONFIG = new Counter.Config(
+                    "platform", "totalNeverSignedStates")
+            .withDescription("total number of states that did not receive enough signatures in the allowed time")
+            .withUnit("count");
+    private final Counter totalNeverSignedStates;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java:48]: You put this metric in the new class, but it's unused here, and still exists in the original `SignedStateMetrics` class

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        writeStateToDiskTime = metrics.getOrCreate(WRITE_STATE_TO_DISK_TIME_CONFIG);
+        totalNeverSignedStates = metrics.getOrCreate(TOTAL_NEVER_SIGNED_STATES_CONFIG);
+        totalNeverSignedDiskStates = metrics.getOrCreate(TOTAL_NEVER_SIGNED_DISK_STATES_CONFIG);
+    }
+
+    /**
+     * Get a metric tracking the average time required to write a state to disk.
+     */
+    @NonNull
+    public RunningAverageMetric getWriteStateToDiskTimeMetric() {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java:75]: The 3 metric getter methods are missing a `@return` tag

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1041]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1048]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java:164]: reformatted

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/DefaultStateSnapshotManager.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/DefaultStateSnapshotManager.java:127]: deduped documentation

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java:48]: Good catch. Technically harmless (second registration gets linked to the same metric), but not good to have. Removed.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java:75]: added

**Review by alittley — APPROVED** [review]

**Comment by lpetrovic05** [comment]
I think package name changes that cause lots of files to be changed should be independent PRs

**Review by jeromy-cannon — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!

**Pull request merged**